prow/git/v2/interactor.go

```diff
@@ -278,6 +282,32 @@ func (i *interactor) FetchRef(refspec string) error {
 	return nil
 }
 
+// FetchFromRemote fetches all update from a specific remote and branch and leaves it as FETCH_HEAD.
+func (i *interactor) FetchFromRemote(remote RemoteResolver, branch string) error {
+	r, err := remote()
+	if err != nil {
+		return fmt.Errorf("couldn't get remote: %v", err)
+	}
+
+	// Parse the remote's URI to censor any existing tokens.
+	u, err := url.Parse(r)
```

Comment:
IMO this should be on the user to do correctly by adding their tokens to the secret agent at startup